sa/model.go

```diff
@@ -12,23 +12,137 @@ import (
 	"github.com/letsencrypt/boulder/core"
 	"github.com/letsencrypt/boulder/features"
 	"github.com/letsencrypt/boulder/probs"
+	"github.com/letsencrypt/boulder/revocation"
 )
 
-const (
-	regV1Fields        string = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol"
-	regV2Fields        string = regV1Fields + ", status"
-	pendingAuthzFields string = "id, identifier, registrationID, status, expires, combinations, LockCol"
-	authzFields        string = "id, identifier, registrationID, status, expires, combinations"
-	sctFields          string = "id, sctVersion, logID, timestamp, extensions, signature, certificateSerial, LockCol"
+type dbSelectOne func(interface{}, string, ...interface{}) error
```

Comment:
I think it might be more natural to define a `type dbSelector interface { Select(...) }`. That way we can just pass a `dbMap` into this. Otherwise, I look at a call like `sa.SelectCertificates(dbMap.Select, ...)` and wonder: What else could I pass instead of `.Select`? Could I pass `.Exec`? What effect would it have?